### ISSUE TITLE
Rewritten Message Not Pushed to Originating Client Connection

### ISSUE THREAD

**Issue by philmaker1** [hoxton-one/deepstream.io#24]
In deepstream version 0.3.1:

When I rewrite a record message inside canPerformAction, the rewritten record data is seen by all clients except for the originating client. The only way to see the updated record is for the originating client to login again. I've also tried without the redis cache and message connectors.

For example, I'm inserting username into a record inside of canPerformAction. Regardless of whether I subscribe to record changes or not, I still don't see the username on the originating client.

Context:
https://github.com/philmaker1/deepstream.io-starter/blob/master/server/scripts/permission/RecordPermissionHandler.js#L35
https://github.com/philmaker1/deepstream.io-starter/blob/master/client/scripts/Pages.js#L400

**Comment by WolframHempel** [comment]
Hi,

transforming data before it leaves the server is a common requirement and we're currently working on a proper solution, rather than the current canPerformAction approach. It will work by providing a list of data transform functions that will be executed for specific actions, e.g. adding a discount for a specific user to an outgoing RPC call.

```javascript
server.set( 'dataTransforms', [{
	topic: C.TOPIC.RPC,
	action: C.ACTIONS.REQUEST,
	transform: function( data, metaData ) {
		data.discount = userDiscounts[ metaData.sender ];
		return data;
	}
}]);
```

This will be released before the end of this month. I appreciate that this doesn't fix your immediate problem, but will hopefully provide a way nicer API going forward. I'd very much appreciate any feedback.

**Comment by WolframHempel** [comment]
Just released deepstream version 0.4.0 with proper support for data-transformations. Please have a look at [http://deepstream.io/tutorials/transforming-data.html](http://deepstream.io/tutorials/transforming-data.html) and let me know if I can close the pull request.
